File: ip-validator/python/ip_validator.py
class IpValidator:
    def validate_ipv4_address(self, ip_address):
        if not ip_address or not isinstance(ip_address, str):
            return False

        parts = ip_address.split('.')
        if len(parts) != 4:
            return False

        for part in parts:
            if not part.isdigit() or not 0 <= int(part) <= 255:
                return False

            if len(part) > 1 and part.startswith('0'):
                return False

        # Check if the last part is '0' or '255'
        return not (parts[3] == '0' or parts[3] == '255')

File: ip-validator/C#/IpValidator/IpValidator.cs
using System;
using System.Linq;

namespace Validator
{
    public class IpValidator
    {
        public bool ValidateIpv4Address(string ipAddress)
        {
            if (string.IsNullOrWhiteSpace(ipAddress))
                return false;

            var parts = ipAddress.Split('.');
            if (parts.Length != 4)
                return false;

            foreach (var part in parts)
            {
                if (!byte.TryParse(part, out byte bytePart))
                    return false;

                if (part.Length > 1 && part.StartsWith("0"))
                    return false;
            }

            // Check if the last part is "0" or "255"
            return !(parts[3] == "0" || parts[3] == "255");
        }
    }
}
